demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1035 @@
+    #error "Define the config democonfigHOSTNAME by following the instructions in file demo_config.h."
+#endif
+
+#if !defined( democonfigENDPOINT ) && defined( democonfigENABLE_DPS_SAMPLE )
+    #error "Define the config dps endpoint by following the instructions in file demo_config.h."
+#endif
+
+#ifndef democonfigROOT_CA_PEM
+    #error "Please define Root CA certificate of the IoT Hub(democonfigROOT_CA_PEM) in demo_config.h."
+#endif
+
+#if defined( democonfigDEVICE_SYMMETRIC_KEY ) && defined( democonfigCLIENT_CERTIFICATE_PEM )
+    #error "Please define only one auth democonfigDEVICE_SYMMETRIC_KEY or democonfigCLIENT_CERTIFICATE_PEM in demo_config.h."
+#endif
+
+#if !defined( democonfigDEVICE_SYMMETRIC_KEY ) && !defined( democonfigCLIENT_CERTIFICATE_PEM )
+    #error "Please define one auth democonfigDEVICE_SYMMETRIC_KEY or democonfigCLIENT_CERTIFICATE_PEM in demo_config.h."
+#endif
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The maximum number of retries for network operation with server.
+ */
+#define sampleazureiotRETRY_MAX_ATTEMPTS                  ( 5U )
+
+/**
+ * @brief The maximum back-off delay (in milliseconds) for retrying failed operation
+ *  with server.
+ */
+#define sampleazureiotRETRY_MAX_BACKOFF_DELAY_MS          ( 5000U )
+
+/**
+ * @brief The base back-off delay (in milliseconds) to use for network operation retry
+ * attempts.
+ */
+#define sampleazureiotRETRY_BACKOFF_BASE_MS               ( 500U )
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milliseconds.
+ */
+#define sampleazureiotCONNACK_RECV_TIMEOUT_MS             ( 10 * 1000U )
+
+/**
+ * @brief The model id for this device
+ *
+ * https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json
+ *
+ * The model id is the JSON document (also called the Digital Twins Model Identifier or DTMI) which
+ * defines the capability of your device. The functionality of the device should match what is
+ * described in the corresponding DTMI. Should you choose to program your own PnP capable device,
+ * the functionality would need to match the DTMI and you would need to update the below 'model_id'.
+ * Please see the sample README for more information on this DTMI.
+ *
+ */
+#define sampleazureiotMODEL_ID                            "dtmi:com:example:Thermostat;1"
+
+/**
+ * @brief Date-time to use for the model id
+ */
+#define sampleazureiotDATE_TIME_FORMAT                    "%Y-%m-%dT%H:%M:%S.000Z"
+
+/**
+ * @brief Telemetry values
+ */
+#define sampleazureiotTELEMETRY_NAME                      "temperature"
+
+/**
+ * @brief Property Values
+ */
+#define sampleazureiotPROPERTY_STATUS_SUCCESS             200
+#define sampleazureiotPROPERTY_SUCCESS                    "success"
+#define sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT    "targetTemperature"
+#define sampleazureiotPROPERTY_MAX_TEMPERATURE_TEXT       "maxTempSinceLastReboot"
+static uint8_t ucPropertyPayloadBuffer[ 256 ];
```

Comment:
how about structuring the file with following format:

#include


#define

export

static

static fn()

fn()

?

Reply:
Here I think it's nice bunching the values together that are related to a specific feature. At a glance, you can see all the things being used for a specific feature. If there is significant passion for grouping by var type I could be swayed but I think this keeps things nicely organized

Reply:
its bit out of sync with our current formatting